File: OCNet/dataset/createList.m
function [] = createList(split, imgDir, varargin)
% Create list for a given dataset. Images and labels must have the same
% name, while images have .jpg extension and labels have the .png one.
% Launch the script from the current directory to get well-organized paths.
% Inputs:
%   - split:    name of the split
%   - imgDir:   path to images
%   - labDir:   path to annotation (optional)
%--------------------------------------------------------------------------
    narginchk(2,3);
    listDir = fullfile(pwd, 'list');
    if ~isempty(varargin)
        labDir = varargin{1};
    else
        labDir = varargin;
    end
    
    imds = imageDatastore(imgDir);
    num_img = numel(imds.Files);

    if ~isempty(labDir)
        lbds = imageDatastore(labDir);
        if num_img ~= numel(lbds.Files)
            error('Wrong number of images');
        end
    end

    % open new file
    splitListFile = fullfile(listDir, [split '.lst']);
    splitList = fopen(splitListFile, 'w+');
    for i=1 : num_img
        if ~isempty(labDir)
            fprintf(splitList, '%s %s\n', imds.Files{i}, lbds.Files{i});
        else    
            fprintf(splitList, '%s\n', imds.Files{i});
        end
    end
    fclose(splitList);
end

File: OCNet/dataset/create_list.py
# Create list file for a given dataset. Images and labels must have the 
# same name, while images have .jpg extension and labels have the .png one.
#------------------------------------------------------------------------
import argparse
import os 

parser = argparse.ArgumentParser()
parser.add_argument('--name', required=True, type=str,
                    help='Name of the split to be built')
parser.add_argument('--img_dir', required=True, type=str,
                    help='Path to the images folder')
parser.add_argument('--lab_dir', required=False, type=str,
                    help='Path to the annotation folder')
args = parser.parse_args()

num_img = len(os.listdir(args.img_dir))

if num_img == 0:
  raise ValueError('%s is empty' %(args.img_dir))
if args.lab_dir is not None:
  num_lab = len(os.listdir(args.lab_dir))
  if num_img != num_lab:
    raise ValueError('%s does not contain the same objects of %s' % (args.img_dir, args.lab_dir))

if args.img_dir.strip().endswith('/'):
  args.img_dir = args.img_dir.strip()[:-1]
if args.lab_dir is not None:
  if args.lab_dir.strip().endswith('/'):
    args.lab_dir = args.lab_dir.strip()[:-1]

list_name = '%s/list/%s.lst' % (os.getcwd(), args.name)
images = os.listdir(args.img_dir)
images.sort() # sort alphabetically 

if args.lab_dir is not None:
  labels = os.listdir(args.lab_dir)
  labels.sort()
  with open(list_name, 'w') as lf:
    for (x,y) in zip(images, labels):
      lf.write('%s/%s %s/%s\n' % (args.img_dir, x, args.lab_dir, y))
else:
  with open(list_name, 'w') as lf:
    for x in images:
      lf.write('%s/%s\n' % (args.img_dir, x))

print('Got %d examples listed at %s' % (num_img, list_name))